docs_user/assemblies/assembly_prepare_director_operator_and_rhoso_for_adoption_process.adoc

```diff
@@ -0,0 +1,192 @@
+ifdef::context[:parent-context: {context}]
+
+[id="prepare-director-operator-and-rhoso-for-adoption-process_{context}"]
+
+:context: prepare-director-operator
+
+= Preparing director Operator for adoption
+
+[NOTE]
+====
+The following procedure uses a single {rhocp_long} cluster that is shared by the source OSPdO environment and a destination {rhos_acro} environment. Two {OpenShiftShort} master nodes are transferred to the {rhos_acro} environment, leaving OSPdO with one node for the duration of the adoption process. After adoption, the remaining node can be used by the {rhos_acro}  environment as well.
+====
+
+Before you adopt your Red Hat OpenStack Platform (RHOSP) director Operator (OSPdO) deployment, you must prepare the Controller nodes for adoption to the RHOSO environment.
+
+. Log in to the {OpenShift } environment where {OpenStackShort} OSPdO is deployed and change to the project that hosts your {OpenStackShort} deployment:
++
+----
+$ oc project openstack
+----
+
+. Define the common environment variables:
+----
+PASSWORD_FILE="tripleo-passwords.yaml"
+export PASSWORD_FILE
+
+OSP18_NAMESPACE="osp18"
+export OSP18_NAMESPACE
+
+OSPDO_NAMESPACE="openstack"
+export OSPDO_NAMESPACE
+
+OS_CLIENT="oc exec -c openstackclient -t openstackclient -- "
+export OS_CLIENT
+
+CONTROLLER_SSH="oc rsh -n $OSPDO_NAMESPACE -c openstackclient openstackclient ssh controller-0.ctlplane"
+export CONTROLLER_SSH
+
+CONTROLLER1_SSH="oc -n $OSPDO_NAMESPACE rsh -c openstackclient openstackclient ssh controller-0.ctlplane"
+export CONTROLLER1_SSH
+CONTROLLER2_SSH=
+export CONTROLLER2_SSH
+CONTROLLER3_SSH=
+export CONTROLLER3_SSH
+
+OSPDO_INTERNAL_API_NET="internalapi"
+export OSPDO_INTERNAL_API_NET
+
+STORAGE_CLASS=$(oc get -n $OSPDO_NAMESPACE pvc openstackclient-hosts -o jsonpath='{.spec.storageClassName}')
+export STORAGE_CLASS
+----
+
+. Get the name of the node where the current {OpenStackShort} controller VM is running:
++
+----
+$ CONTROLLER_NODE=$(oc -n $OSPDO_NAMESPACE get vmi -ojson | jq -r '.items[0].status.nodeName')
+export CONTROLLER_NODE
+----
++
+
+
+In case we're using a combo master/worker cluster with an HA OSPdO env,
+configure OSPdO to use a single Osp controller:
+
+Disable stonith:
+[,bash]
+----
+stonith_resources=$(sudo pcs stonith status|grep Started|awk '{print $2}')
+for in in stonith_resources ;do CONTROLLER1_SSH sudo pcs stonith disable ;done
+----
+Put non controller nodes in maintannace
+[,bash]
+----
+CONTROLLER1_SSH sudo pcs node maintenance controller-1 controller-2
+----
```

Comment:
```suggestion
.. Put non-Controller nodes in maintenance:
+
----
$CONTROLLER1_SSH sudo pcs node maintenance controller-1 controller-2
----
```

Reply:
I think this is a good example of needing a user-replacement value. I could see customers blindly copying and pasting "controller-1, controller-2".

Reply:
ok